### PR TITLE
Category storage/partiql permissions

### PR DESCRIPTION
**Pull request by johnf**
#### Description of changes

Add PartiQL permissions to IAM policy.

Note I've also done my best to lint the file. I've ignored some of the errors where the complexity out strips my knowledge of the project.

#### Issue #, if available

#8430

#### Description of how you validated changes

Visual inspection. It is a very simple change and maps the permissions I've added in my production app.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.ts [hunk 1]
```diff
@@ -644,137 +666,142 @@ async function addTrigger(context: $TSContext, resourceName: string, triggerList
...
-  let policy = {};
+};
+
+/**
+ * getIAMPolicies: get the IAM policies for CRUD
+ * @param resourceName - The resourceName
+ * @param crudOptions - list of CRUD actions
+ * @returns DynamoDBCLIInputs
+ */
+export const getIAMPolicies = (resourceName: ResourceName, crudOptions: $TSAny): { policy: $TSObject, attributes: string[] } => {
```
Comment (edwardfoyle) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.ts:758]: For future me or other reviewers, the functional changes in this PR are adding the PartiQL* actions to the policies here

**Comment by edwardfoyle** [comment]
Hi @johnf if you are still interested in contributing this change, could you add a test around the expected actions generated by `getIAMPolicies`?

**Comment by johnf** [comment]
@edwardfoyle Can you point me at some example tests, perhaps in anther package?

Should I be using the walkthrough logic like the tests that already exist in that package or simply testing getIAMPolicies directly?

**Comment by edwardfoyle** [comment]
Just unit tests around the expect output of getIAMPolicies for CRUD operations is sufficient

**Comment by johnf** [comment]
@edwardfoyle tests added

**Review by sobolk — APPROVED** [review]

**Pull request merged**